tests/sessions/claims/test_verify_session.py

```diff
@@ -0,0 +1,453 @@
+        return {"handle": s.get_handle()}
+
+    refetched_claims_verify_session = verify_session(
+        override_global_claim_validators=lambda _, __, ___: [  # type: ignore
+            TrueClaim.validators.has_value(True)
+        ]
+    )
+
+    @app.get("/refetched-claim")
+    async def refetched_claim(  # type: ignore
+        s: SessionContainer = Depends(refetched_claims_verify_session),
+    ):
+        return {"handle": s.get_handle()}
+
+    refetched_claims_false_verify_session = verify_session(
+        override_global_claim_validators=lambda _, __, ___: [  # type: ignore
+            TrueClaim.validators.has_value(False)
+        ]
+    )
+
+    @app.get("/refetched-claim-false")
+    async def refetched_claim2(  # type: ignore
+        s: SessionContainer = Depends(refetched_claims_false_verify_session),
+    ):
+        return {"handle": s.get_handle()}
+
+    class CustomValidator(SessionClaimValidator):
+        def __init__(self, is_valid: bool):
+            super().__init__("test_id")
+            self.is_valid = is_valid
+
+        def validate(
+            self, payload: JSONObject, user_context: Union[Dict[str, Any], None] = None
+        ) -> Union[ClaimValidationResult, Awaitable[ClaimValidationResult]]:
+            if self.is_valid:
+                return {"isValid": True}
+            return {"isValid": False, "reason": "test_reason"}
+
+    refetched_claims_verify_session_is_valid_false = verify_session(
+        override_global_claim_validators=lambda _, __, ___: [  # type: ignore
+            CustomValidator(is_valid=False)
+        ]
+    )
+
+    @app.get("/refetched-claim-isvalid-false")
+    async def refetched_claim3(  # type: ignore
+        s: SessionContainer = Depends(refetched_claims_verify_session_is_valid_false),
+    ):
+        return {"handle": s.get_handle()}
+
+    refetched_claims_verify_session_is_valid_true = verify_session(
+        override_global_claim_validators=lambda _, __, ___: [  # type: ignore
+            CustomValidator(is_valid=True)
+        ]
+    )
+
+    @app.get("/refetched-claim-isvalid-true")
+    async def refetched_claim4(  # type: ignore
+        s: SessionContainer = Depends(refetched_claims_verify_session_is_valid_true),
+    ):
+        return {"handle": s.get_handle()}
+
+    return TestClient(app)
+
+
+def create_session(fastapi_client: TestClient):
+    res = fastapi_client.post("create-with-claim", json={"key": "something"})
+    return res
+
+
+async def test_should_allow_without_claims_required_or_present(
+    fastapi_client: TestClient,
+):
+    st_init_args = {
+        **st_init_common_args,
+        "recipe_list": [session.init(anti_csrf="VIA_TOKEN")],
+    }
+    init(**st_init_args)  # type: ignore
+    start_st()
+
+    create_session(fastapi_client)
+
+    res = fastapi_client.get("/default-claims")
+    assert res.status_code == 200
+    assert "-" in res.json()["handle"]
```

Comment:
I'm not sure what this does? Does it just check if we have the handle in the response body (and so that we have the right body supplied by the user)?

Reply:
Yes. Rest is same as the Node PR..  (`/default-claims` API uses `verify_session`)

Reply:
Hmm. It's a bit strange, but OK. I'd prefer if it didn't rely on the format of the session handle but I guess it's fine.